src/rules/unnecessary_function_arguments.erl

```diff
@@ -160,18 +183,14 @@ is_clause_a_nif_stub(Clause) ->
 %% @doc Checks if the given function node implements a callback
 -spec is_callback(erl_syntax:syntaxTree(), string(), imp_callbacks()) -> boolean().
 is_callback(FunctionNode, File, ImpCallbacks) ->
-    case maps:get(File, ImpCallbacks, []) of
-        Callbacks when is_list(Callbacks) ->
-            lists:member(
-                hank_utils:function_tuple(FunctionNode), Callbacks);
-        _ ->
-            false
-    end.
+    lists:member(
+        hank_utils:function_tuple(FunctionNode), maps:get(File, ImpCallbacks, [])).
```

Comment:
We only have lists at this point… `syntax_error` is discarded by `is_parseable/2`.